iOS/rainbow/Controller/Camera/CameraController.swift

```diff
@@ -8,12 +8,48 @@
 
 import Foundation
 import Lumina
+import SpriteKit
+import AudioToolbox
+
+enum GameCameraState {
+    case shouldStartNewGame // when the game should be started, or has finished and could be restarted
+    case nothingDetected // normal camera state, hunting for objects
+    case detectionInProgress // it's fair to assume they should detect three consecutive frames since this happens fast
+    case objectDetected // this should last one second when the fireworks are going off
+}
+
+enum RainbowDetectedObject: String {
```

Comment:
there's a lot of hardcoded "if apple, then this; if bee, then that" -- i'm trying to think how to best allow a developer creating their own game to proceed with ease, hear me out.

how about placing the object in a config file, like this? (and then read more comments)

```
"apple": "path_to_image"
...
```